packages/flutter/lib/src/painting/continuous_stadium_border.dart

```diff
@@ -0,0 +1,357 @@
+// Copyright 2018 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:math' as math;
+
+import 'basic_types.dart';
+import 'borders.dart';
+import 'edge_insets.dart';
+
+/// A stadium border with continuous corners.
+///
+/// A shape similar to a stadium, but with a smoother transition from
+/// each linear edge to its 180º curves. Each 180º curve is approximately half
+/// an ellipse.
+///
+/// In this shape, the curvature of each 180º curve around the length of the arc
+/// is approximately a gaussian curve instead of a step function as with a
+/// traditional half circle round.
+///
+/// The ~3 represents twice the ratio (ie. ~3/2) of a corner's declared radius
```

Comment:
I think this was copy pasted but doesn't fit in the flow of the paragraphs here. Make sure this doc reads right.